src/families/cosmos/js-synchronisation.ts

```diff
@@ -42,56 +42,87 @@ const txToOps = (info: any, id: string, txs: any): Operation[] => {
-
-          if (attributes["sender"] === address) {
-            op.type = "OUT";
-            op.value = op.value.plus(fees);
-          } else if (attributes["recipient"] === address) {
-            op.type = "IN";
+          if (
+            attributes["sender"] &&
+            attributes["recipient"] &&
+            attributes["amount"]
+          ) {
+            op.senders.push(attributes["sender"]);
+            op.recipients.push(attributes["recipient"]);
+
+            if (attributes["amount"].indexOf(currency.units[1].code) != -1) {
+              op.value = op.value.plus(
+                attributes["amount"].replace(currency.units[1].code, "")
+              );
+            }
+
+            if (attributes["sender"] === address) {
+              op.type = "OUT";
+              op.value = op.value.plus(fees);
+            } else if (attributes["recipient"] === address) {
+              op.type = "IN";
+            }
           }
           break;
 
         case "withdraw_rewards":
-          op.type = "REWARD";
-          op.value = new BigNumber(fees);
-          op.extra.validators.push({
-            amount: attributes["amount"].replace(currency.units[1].code, ""),
-            address: attributes.validator,
-          });
+          if (
+            attributes["amount"] &&
+            attributes["amount"].indexOf(currency.units[1].code) != -1
+          ) {
+            op.type = "REWARD";
+            op.value = new BigNumber(fees);
+            op.extra.validators.push({
+              amount: attributes["amount"].replace(currency.units[1].code, ""),
+              address: attributes.validator,
+            });
+          }
           break;
 
         case "delegate":
-          op.type = "DELEGATE";
-          op.value = new BigNumber(fees);
-          op.extra.validators.push({
-            amount: attributes["amount"].replace(currency.units[1].code, ""),
-            address: attributes.validator,
-          });
+          if (
+            attributes["amount"] &&
+            attributes["amount"].indexOf(currency.units[1].code) != -1
+          ) {
+            op.type = "DELEGATE";
+            op.value = new BigNumber(fees);
+            op.extra.validators.push({
+              amount: attributes["amount"].replace(currency.units[1].code, ""),
+              address: attributes.validator,
+            });
+          }
           break;
 
         case "redelegate":
-          op.type = "REDELEGATE";
-          op.value = new BigNumber(fees);
-          op.extra.validators.push({
-            amount: attributes["amount"].replace(currency.units[1].code, ""),
-            address: attributes.destination_validator,
-          });
-          op.extra.cosmosSourceValidator = attributes.source_validator;
+          if (
+            attributes["amount"] &&
+            attributes["amount"].indexOf(currency.units[1].code) != -1 &&
+            attributes.destination_validator &&
+            attributes.source_validator
```

Comment:
We should avoid mixing styles, `attributes.xxx` should be preferred over `attributes["xxx"]` where possible.

Reply:
fix in 892148b6dc81a8928bb10c76699f4bd6f00c2bf9